kyo-core/shared/src/main/scala/kyo/layers.scala

```diff
@@ -0,0 +1,131 @@
+package kyo
+
+object layers {
+
+  trait Layer[In, Out] { self =>
+    def run[T, S](effect: T > (S with In))(implicit fl: Flat[T > (S with In)]): T > (S with Out)
```

Comment:
In Scala 2, methods that remove items from the pending effects need to have the effect to be removed as the first in the type intersection otherwise the compiler isn't able to infer types correctly. You might be able to avoid type annotations if you use `effect: T > (In with S)`.

Reply:
I see. It could be that the compiler isn't able to infer the removal based on the generic type parameter.